### PR TITLE
Monthly Financial Reporting - Office/Exam Group Filtering

### PR DESCRIPTION
**Pull request by akroon3r**
A CSR/GA/Liaison will require the ability to generate financial reports based on type of exam as well as for a specific office. This PR contains the added functionality to add an exam_type and office_name url parameter to the url used to export a csv to the clients browser. At this moment in time, both parameters are not required or used when creatiung the CSV, but will be required in future development when generating these csv files.

### CODE REVIEW

File: api/app/resources/bookings/booking/booking_post.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ class BookingPost(Resource):
     def post(self):
 
         csr = CSR.find_by_username(g.oidc_token_info['username'])
+        print ("CSR: ", csr)
```
Comment (akroon3r) [inline api/app/resources/bookings/booking/booking_post.py]: getting rid of this

File: frontend/src/exams/generate-financial-report-modal.vue [hunk 2]
```diff
@@ -16,47 +16,99 @@
...
             let form_end_date = this.endDate
-            let url= '/exams/export/?start_date=' + form_start_date + '&end_date=' + form_end_date
+            let exam_type = this.selectedExamType
+            let office_name = this.selectedOffice
+            let url= '/exams/export/?start_date=' + form_start_date + '&end_date=' + form_end_date + '&exam_type='
+                + exam_type + '&office_name=' + office_name
+            let today = moment().format('YYYY_MM_DD-HH:MM:SS')
+            let filename = 'export-csv-' + today
             this.getExamsExport(url)
                 .then(resp => { const FileDownload = require('js-file-download')
```
Comment (sjrumsby) [inline frontend/src/exams/generate-financial-report-modal.vue]: Some minor formatting issues here

File: frontend/src/exams/generate-financial-report-modal.vue [hunk 3]
```diff
@@ -16,47 +16,99 @@
       </b-row>
       <b-row class="my-1">
         <b-col sm="4"><label>Start Date:</label></b-col>
-        <b-col sm="6"><b-form-input id="startDate" type="date" v-model=startDate></b-form-input></b-col>
+        <b-col sm="6">
+          <b-form-input id="startDate"
+                        type="date"
+                        v-model=startDate />
+          <b-form-invalid-feedback></b-form-invalid-feedback>
```
Comment (sjrumsby) [inline frontend/src/exams/generate-financial-report-modal.vue]: Does this `form-invalid-feedback` tag show errors with both start and end dates? If so, should it be at the top of the form above all inputs?

File: frontend/src/exams/generate-financial-report-modal.vue [hunk 4]
```diff
@@ -16,47 +16,99 @@
...
-
-            console.log("Pressed Submit")
             let form_start_date = this.startDate
             let form_end_date = this.endDate
-            let url= '/exams/export/?start_date=' + form_start_date + '&end_date=' + form_end_date
+            let exam_type = this.selectedExamType
+            let office_name = this.selectedOffice
+            let url= '/exams/export/?start_date=' + form_start_date + '&end_date=' + form_end_date + '&exam_type='
+                + exam_type + '&office_name=' + office_name
+            let today = moment().format('YYYY_MM_DD-HH:MM:SS')
```
Comment (sjrumsby) [inline frontend/src/exams/generate-financial-report-modal.vue]: I believe colons are an invalid filename in windows. If we're doing ISO 8601 then something like `YYYY-MM-DDTHHMMSS` should work. Not sure how moment plays with that `T` - might need to do `YYYY-MM-DD_HHMMSS`.

File: frontend/src/exams/generate-financial-report-modal.vue [hunk 5]
```diff
@@ -16,47 +16,99 @@
...
+            this.endDate = ''
+            this.selectedExamType = ''
+            this.setSelectedOffice('')
           },
         },
         computed: {
           ...mapState({
             showGenFinReportModal: state => state.showGenFinReportModal,
+            offices: state => state.offices,
+            capturedExam: state => state.capturedExam,
```
Comment (sjrumsby) [inline frontend/src/exams/generate-financial-report-modal.vue]: Is `capturedExam` still used here?

File: frontend/src/exams/generate-financial-report-modal.vue (hunk 2 above)
Comment (akroon3r) [reply frontend/src/exams/generate-financial-report-modal.vue]: screen shot 2019-01-31 at 9 30 18 am

File: frontend/src/exams/generate-financial-report-modal.vue (hunk 3 above)
Comment (akroon3r) [reply frontend/src/exams/generate-financial-report-modal.vue]: nope this was an attempt at form validation before we talked on tuesday, and forgot to remove them. thanks for catching that, they've been removed.

File: frontend/src/exams/generate-financial-report-modal.vue (hunk 4 above)
Comment (akroon3r) [reply frontend/src/exams/generate-financial-report-modal.vue]: fixed.

File: frontend/src/exams/generate-financial-report-modal.vue (hunk 5 above)
Comment (akroon3r) [reply frontend/src/exams/generate-financial-report-modal.vue]: nope, removed.

**Pull request merged**